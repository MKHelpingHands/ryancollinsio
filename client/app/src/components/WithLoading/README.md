## WithLoading Component
A component that ...

### Example

```js
<WithLoading />
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **myProp**    | String   |             | Any string value


### Other Information
