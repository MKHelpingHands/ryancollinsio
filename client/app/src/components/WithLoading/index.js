import React, { PropTypes } from 'react';
import { LoadingIndicator } from 'components';
import Section from 'grommet-udacity/components/Section';

const WithLoading = ({
  isLoading,
  fullscreen,
  children,
}) => (
  <div>
    {isLoading &&
      <Section
        className={fullscreen ? 'full-height' : ''}
        justify="center"
        align="center"
      >
        <LoadingIndicator />
      </Section>
    }
    {!isLoading &&
      <div>
        {children}
      </div>
    }
  </div>
);

WithLoading.propTypes = {
  children: PropTypes.node.isRequired,
  isLoading: PropTypes.bool.isRequired,
  fullscreen: PropTypes.bool.isRequired,
};

WithLoading.defaultProps = {
  fullscreen: true,
  isLoading: true,
};

export default WithLoading;
