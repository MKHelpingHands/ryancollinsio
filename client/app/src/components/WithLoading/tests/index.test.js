import WithLoading from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<WithLoading />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <WithLoading />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
