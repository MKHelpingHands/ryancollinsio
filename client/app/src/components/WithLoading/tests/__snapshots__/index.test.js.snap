exports[`<WithLoading /> should render with default props 1`] = `
<div>
  <Section
    align="center"
    className="full-height"
    justify="center"
    pad={
      Object {
        "vertical": "medium"
      }
    }>
    <undefined />
  </Section>
</div>
`;
